### PR TITLE
refactor: remove associated error types from internal traits

### PR DESCRIPTION
**Pull request by djordon**
## Description

Implements part of the work done in https://github.com/stacks-network/sbtc/pull/507.

## Changes

* Remove the associated error type from the `DbRead`, `DbWrite`, `BitcoinInteract`, and `MessageTransfer` traits.
* Fixes bug/typo in the tests that may have caused tests in `main` to fail.

## Testing Information

This is just a refactor, so if the tests pass then all should be well.

## Checklist:

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: signer/src/network/in_memory.rs [hunk 1]
```diff
@@ -55,32 +57,21 @@ impl Default for Network {
 }
 
 impl super::MessageTransfer for MpmcBroadcaster {
-    type Error = Error;
-    async fn broadcast(&mut self, msg: super::Msg) -> Result<(), Self::Error> {
+    async fn broadcast(&mut self, msg: super::Msg) -> Result<(), Error> {
         self.recently_sent.push_back(msg.id());
-        self.sender.send(msg).map_err(|_| Error::Send)?;
+        self.sender.send(msg).unwrap();
```
Comment (cylewitruk) [inline signer/src/network/in_memory.rs:62]: Are we okay with using unwraps here? Thinking if we ever did enable using the in-memory db at runtime (for regtest/testnet etc.)? I'm not against it, just that we have a decision that unwraps are OK for the in-mem DB.

**Review by cylewitruk — COMMENTED** [review]
Ahhh, thank you for this! :clinking_glasses: Just causes headaches when we know what error type we're going to be using lol.. Just had one Q about the unwraps in the in-mem db.

File: signer/src/network/in_memory.rs (hunk 1 above)
Comment (djordon) [reply signer/src/network/in_memory.rs:62]: I don't know. I'm generally of the opinion unwraps aren't okay, but this is extremely  close to a testing utility. And unlike our in-memory db, I haven't used this stuff at all, so I had half a mind to remove it. 

But thinking about this some more, I'll switch back to using our normal pattern and later we can decide on what, if anything, to do with this module.

Comment (cylewitruk) [reply signer/src/network/in_memory.rs:62]: Sounds good :+1:

**Review by cylewitruk — APPROVED** [review]
:heavy_check_mark: lgtm

**Pull request merged**